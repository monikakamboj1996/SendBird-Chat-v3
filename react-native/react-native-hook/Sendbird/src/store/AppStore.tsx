import AsyncStorage from '@react-native-async-storage/async-storage';
import { createStore, applyMiddleware } from 'redux';

import { persistStore, persistReducer } from 'redux-persist';
import thunk from 'redux-thunk';
import RootReducer from '../reducer/RootReducer';

// Middleware: Redux Persist Config
const persistConfig = {
  // Root
  key: 'root',
  // Storage Method (React Native)
  storage: AsyncStorage,
};

// Middleware: Redux Persist Persisted Reducer
const persistedReducer = persistReducer(persistConfig, RootReducer);

// Redux: Store
const store = createStore(persistedReducer, applyMiddleware(thunk));

// Middleware: Redux Persist Persister
let persistor = persistStore(store);
// Exports
export { store, persistor };
