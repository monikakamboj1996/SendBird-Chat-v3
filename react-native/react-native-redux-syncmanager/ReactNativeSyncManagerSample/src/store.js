import {createStore, compose, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import reducers from './reducer';

export const store = createStore(reducers, {}, compose(applyMiddleware(thunk)));
