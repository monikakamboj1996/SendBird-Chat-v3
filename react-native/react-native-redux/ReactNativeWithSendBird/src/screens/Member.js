import React, { Component } from 'react';
import { View, FlatList, Text } from 'react-native';
import { connect } from 'react-redux';
import { initMember, getParticipantList, getMemberList } from '../actions';
import { Button, Spinner, ListItem, Avatar } from '../components';
import { sbCreateParticipantListQuery } from '../sendbirdActions';

class Member extends Component {
  static navigationOptions = ({ navigation }) => {
    const { params } = navigation.state;
    return {
      title: params.isOpenChannel ? 'Participants' : 'Members',
      headerLeft: (
        <Button
          containerViewStyle={{ marginLeft: 0, marginRight: 0 }}
          buttonStyle={{ paddingLeft: 14 }}
          icon={{
            name: 'chevron-left',
            type: 'font-awesome',
            color: '#7d62d9',
            size: 18
          }}
          backgroundColor="transparent"
          onPress={() => navigation.goBack()}
        />
      )
    };
  };

  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
      participantListQuery: null,
      list: []
    };
  }

  componentDidMount() {
    this._initMember();
  }

  componentDidUpdate(prevProps, prevState, snapshot) {
    const { list } = this.props;

    if (list !== prevProps.list) {
      const newList = [...this.state.list, ...list];
      this.setState({
        isLoading: false,
        list: newList
      });
    }
  }

  _initMember = () => {
    this.props.initMember();
    this._getMemberList(true);
  };

  _getMemberList = init => {
    const { isOpenChannel } = this.props.navigation.state.params;
    if (isOpenChannel) {
      this._getOpenChannelParticipantList(init);
    } else {
      this._getGroupChannelMemberList();
    }
  };

  _getOpenChannelParticipantList = init => {
    if (!init && !this.state.participantListQuery) {
      return;
    }
    const { isOpenChannel, channelUrl } = this.props.navigation.state.params;
    this.setState({ isLoading: true }, () => {
      if (init) {
        sbCreateParticipantListQuery(channelUrl)
          .then(participantListQuery => {
            this.setState({ participantListQuery }, () => {
              this.props.getParticipantList(this.state.participantListQuery);
            });
          })
          .catch(error => this.props.navigation.goBack());
      } else {
        this.props.getParticipantList(this.state.participantListQuery);
      }
    });
  };

  _getGroupChannelMemberList = () => {
    const { channelUrl } = this.props.navigation.state.params;
    this.setState({ isLoading: true, list: [] }, () => {
      this.props.getMemberList(channelUrl);
    });
  };

  _renderList = ({ item }) => {
    return (
      <ListItem
        containerStyle={{ backgroundColor: '#fff' }}
        avatar={
          <Avatar rounded source={item.profileUrl ? { uri: item.profileUrl } : require('../img/icon_sb_68.png')} />
        }
        title={item.nickname}
        titleStyle={{ fontWeight: '500', fontSize: 16 }}
        rightTitle={item.isOnline}
        rightTitleStyle={{
          color: item.isOnline === 'online' ? '#37b24d' : '#878D99'
        }}
        rightIcon={<Text></Text>}
      />
    );
  };

  render() {
    return (
      <View>
        <Spinner visible={this.state.isLoading} />
        <FlatList
          renderItem={this._renderList}
          data={this.state.list}
          onEndReached={() => this._getMemberList(false)}
          onEndReachedThreshold={-50}
          keyExtractor={(item, index) => index.toString()}
        />
      </View>
    );
  }
}

function mapStateToProps({ member }) {
  const { list } = member;
  return { list };
}

export default connect(
  mapStateToProps,
  {
    initMember,
    getParticipantList,
    getMemberList
  }
)(Member);
