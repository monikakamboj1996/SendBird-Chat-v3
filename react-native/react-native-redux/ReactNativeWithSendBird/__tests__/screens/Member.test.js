import 'react-native';
import React from 'react';
import Member from '../../src/screens/Member';
import store from '../../src/store';
import renderer from 'react-test-renderer';

const navigation = {
  setParams: jest.fn(),
  navigate: jest.fn(),
  state: {
    params: {
      channelUrl: 'channelUrl',
      isOpenChannel: true
    }
  },
  dispatch: jest.fn(),
  addListener: jest.fn()
};

describe('screen/Member', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Member store={store} navigation={navigation} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // TODO: add test cases
});
