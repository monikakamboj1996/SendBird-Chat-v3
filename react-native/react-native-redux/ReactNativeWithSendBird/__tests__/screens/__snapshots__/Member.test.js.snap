// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen/Member renders correctly 1`] = `
<View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "landscape",
        "portrait",
      ]
    }
    transparent={true}
    visible={true}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "flex": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.25)",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="white"
          hidesWhenStopped={true}
          size="large"
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {},
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "bottom": 0,
                "flex": 1,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "height": 50,
                  "top": 80,
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </Modal>
  <RCTScrollView
    dataSource={
      ListViewDataSource {
        "items": 0,
      }
    }
    enableEmptySections={true}
    onEndReached={[Function]}
    onEndReachedThreshold={-50}
    renderRow={[Function]}
    renderScrollComponent={[Function]}
  >
    <View />
  </RCTScrollView>
</View>
`;
